### PR TITLE
Feat(eos_cli_config_gen): Add CRL support for management security

### LINKED ISSUE #3380: Feat(eos_cli_config_gen): Add support for `crl` under management_security.ssl_profiles
### Enhancement summary

```
management security
          ssl profile ProfileA
              certificate ProfileA.crt key ProfileA.key
              trust certificate aristaDeviceCertProvisionerDefaultRootCA.crt
              crl aristaDeviceCertProvisionerDefaultRootCA.crl.  <<<<<<<<<<<<<<<<<<<<<
```

```
management_security:
  ssl_profiles:
    - name: <profile_name>
       crl: ... 
```

as part of this issue, validate support for the other keys and make sure it can generate the required config

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

being able to revoke certs

### Describe the solution you would like

cf summary

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by harshitk-arista**
## Change Summary

Allow specifying a list of CRLs for an SSL profile

## Related Issue(s)

Fixes #3380

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
CF

## How to test

Molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md [hunk 1]
```diff
@@ -109,4 +111,9 @@
           certificate:
             file: <str>
             key: <str>
+
+          # List of CRLs (Certificate Revocation List).
+          # One CRL needs to be specified for every CA in the chain, even if it's not revoking any certificate.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md]: I am not sure this is true. The revocation lists are not required on EOS are they?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_security.schema.yml [hunk 2]
```diff
@@ -154,3 +154,10 @@ keys:
                   type: str
                 key:
                   type: str
+            certificate_revocation_lists:
+              type: list
+              description: |
+                List of CRLs (Certificate Revocation List).
+                One CRL needs to be specified for every CA in the chain, even if it's not revoking any certificate.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_security.schema.yml]: Not sure I understand what this means exactly... There should be only one CA right and then intermatidate CAs maybe? so a config is only valid if there is an entry for every single intermediate CA?
```suggestion
                One CRL needs to be specified for every CA in the chain, even if it is not revoking any certificate.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2 [hunk 3]
```diff
@@ -28,11 +28,19 @@
...
-| SSL Profile Name | TLS protocol accepted | Certificate filename | Key filename | Cipher List |
-| ---------------- | --------------------- | -------------------- | ------------ | ----------- |
+| SSL Profile Name | TLS protocol accepted | Certificate filename | Key filename | Cipher List | CRLs |
+| ---------------- | --------------------- | -------------------- | ------------ | ----------- | ---- |
 {%         set ssl_profiles_certs = [] %}
 {%         for ssl_profile in management_security.ssl_profiles | arista.avd.natural_sort %}
-| {{ ssl_profile.name | arista.avd.default('-') }} | {{ ssl_profile.tls_versions | arista.avd.default('-') }} | {{ ssl_profile.certificate.file | arista.avd.default('-') }} | {{ ssl_profile.certificate.key | arista.avd.default('-') }} | {{ ssl_profile.cipher_list | arista.avd.default('-') }} |
+{%             set crls = "" %}
+{%             if ssl_profile.certificate_revocation_lists is arista.avd.defined %}
+{%                 for crl in ssl_profile.certificate_revocation_lists | arista.avd.natural_sort %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2]: natural_sort will return an empty list if the variable does not exist
```suggestion
{%             for crl in ssl_profile.certificate_revocation_lists | arista.avd.natural_sort %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md (hunk 1 above)
Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md]: This is documented here - https://www.arista.com/en/um-eos/eos-control-plane-security

grep for 'One CRL needs to be specified for every CA'

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2 [hunk 4]
```diff
@@ -28,11 +28,19 @@
...
+{%                     set crls = crls ~ crl ~ "<br>" %}
+{%                 endfor %}
+{%             else %}
+{%                 set crls = "-" %}
+{%             endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2]:
```
{% set crls = ssl_profile.certificate_revocation_lists | arista.avd.natural_sort | join("<br>") %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-security.j2 [hunk 5]
```diff
@@ -92,5 +92,10 @@ management security
 {%         if ssl_profile.certificate is arista.avd.defined %}
       certificate {{ ssl_profile.certificate.file }} key {{ ssl_profile.certificate.key }}
 {%         endif %}
+{%         if ssl_profile.certificate_revocation_lists is arista.avd.defined %}
+{%             for crl in ssl_profile.certificate_revocation_lists | arista.avd.natural_sort %}
+      crl {{ crl }}
+{%             endfor %}
+{%         endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-security.j2]:
```suggestion
{%         for crl in ssl_profile.certificate_revocation_lists | arista.avd.natural_sort %}
      crl {{ crl }}
{%         endfor %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_security.schema.yml (hunk 2 above)
Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_security.schema.yml]: This means every intermediate certificate as well as root certificate.

If a CRL for each is not specified, the SSL profile won't be valid.

Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_security.schema.yml]: Empty does not mean that the crl file can be empty. It needs to be a fully formed CRL in PEM format, just that the revoked certs list in the CRL can be empty.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md (hunk 1 above)
Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md]: Yes the crl field itself is optional. but if it is specified, one for each cert in the chain needs to be specified. Let me rephrase this a bit.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2 (hunk 4 above)
Comment (harshitk-arista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2]: Thanks for this. Much clearer.

**Review by gmuloc — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md]: You can use min_length on the list in the schema to enforce this.

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/management-security.md]: nvm I misunderstood. Resolving.
